cmd/thanos/sidecar.go

```diff
@@ -298,13 +338,28 @@ func runSidecar(
 	return nil
 }
 
+func validatePrometheus(ctx context.Context, logger log.Logger, promURL *url.URL, tsdbPath string) error {
```

Comment:
As @GiedriusS mentioned this should depend IF sidecar is configured to upload old blocks or not.

If upload is disabled compaction SHOULD be enabled even (but does not have to). So this code is wrong.

Also IMO this function is too small to be.. function. It's too shallow. I would rather pack all (check Prom version, skip if not version etc) into `validateProm`

Reply:
Done.